### ISSUE TITLE
AutoSSL failing to renew certs

### ISSUE THREAD

**Issue by michieloosterling** [engintron/engintron#1037]
Hi there,

I need to preface this by mentioning that there are a bunch of closed issues issues pertaining to AutoSSL failures which I've read by now. However, none of those posts have helped me fix the issue I've encountered, the last instance of which was this morning.
I maintain 3 VPSs, only one of which is running Engintron sort of as a test at the moment to be honest I don't trust it.

To start with, here are the highlights of my setup for this particular server:

- VPS running CentOS 7.6.1810
- WHM / cPanel v78.0.20
- 4GB ram
- 2 accounts, one of which is running a WP multisite network using 14 domains with on average at least one domain alias per site. The other account is a standalone site with 1 domain alias.
- AutoSSL uses Comodo for certs (customer did not want Let's Encrypt for reasons beyond the scope of this issue).

On to my issue then. In short, with Engintron enabled, AutoSSL is unable to get new certificates for domains. The AutoSSL logs show issues mutliple issues such as Local HTTP DCV errors and Local DNS DCV errors for the main domain and subdomains (think cpanel.domainname.com type domains, which do exist in the DNS).
The result of which is catastrophic; the domains in question can no longer be reached by visitors as they throw up warnings about certs not existing and not being able to set up a secure connection.
Disabling EnginTron and running AutoSSL instantly results in valid certs being installed. I can enable EnginTron afterwards and the certs keep on working, until one of the domains needs a new one, and the cycle repeats itself.

Now, issue #753 lists some rules to be inserted into the custom rules for Nginx. After the first time AutoSSL failed I found post #753, copied the rules with some very minor modifications (added something to the $host check) into the custom_rules file via the interface in WHM, saved the config and restarted Nginx. I checked afterwards and saw the rules were still there. However, after checking this morning the custom rules were gone, which is of course odd in and by itself.

The issue would seem to be that the contents of the .well_known directory is cached by Engintron and that as a result, when Comodo comes checking, the cached results are served up which obviously wouldn't include any of the very temporary text files that Comodo is looking for for verification.

Secondly I find it odd the custom rules disappeared. The rules in question would seem like they perfectly meet the requirements I have, however with no rules apparently present I can see why AutoSSL failed due to the caching done.

Regardless of fixes for my case, I would suggest rules pertaining to AutoSSL to be included in the 'out of the box' config by default. Might save a lot of headaches.

---------

Either way, as of right now the domains in question are working again with SSL. Engintron is however turned off for the foreseeable future until I can get this resolved. I really don't want to risk getting this particular client more pissed than they already are.

Hopefully someone will be able to assist or at least point me in the right direction. If more information is needed I'll be happy to supply.

**Comment by michieloosterling** [comment]
Anyone have any suggestions at all?
Still haven't turned Engintron back on, and I would actually like to do so!

**Comment by fevangelou** [comment]
First off, you start by saying "to be honest I don't trust it". That's a bad way to start using any software. If you have valid reasons not to trust something (difficult in Engintron's case of course), well, don't use it.

Secondly, failure to renew certificates is one thing and not serving SSL another. The first may be caused by various factors (e.g. site behind CloudFlare). The latter can be caused by cached browser certificates. When switching from Apache to Nginx, use a private browser window to verify a certificate is changed. This is the only true way to know, as browsers cache SSL certs for up to a few mins. If you use HSTS though (bad idea) you're forcing browsers to cache a bad certificate for way longer.

**Comment by michieloosterling** [comment]
Hi, sorry, meant that more as "I installed it to a semi-test environment, it started out great, then it did funky things, and hence I don't trust it enough to deploy it on my two other more important and vital VPSs." Which probably makes more sense. Didn't mean to imply I didn't trust it from the get go, as I did; which is why I chose to test it out to begin with ;)
(English is not my native language, sorry)

With that (hopefully?) out of the way, on to your suggestions.

The server itself nor the sites running on the server are behind any kind of service such as CloudFlare. 
As for the cached certificates, we don't have HSTS enabled. After enabling Engintron I did of course check all sites running on the server, both with and without private windows (and more importantly, from multiple machines and connections), and all reported back fine and worked fine for around 1 1/2 weeks until it came time for AutoSSL to renew one of the certs. 

As I mentioned, it seems the issue lies in the caching of the contents of the .well_known directory, which normally is empty. I assume this is due to Nginx / Engintron, as with Engintron disabled, it worked right away. With Engintron enabled, I see the temporary files needed for HTTP validation being created, however they can't be found via external sources such as the Comodo verification tool. 
With it disabled, the same files are of course created and instantly found by Comodo.

I tried the exclude rules I mentioned in my original post, but those have mysteriously disappeared from the config.

So, is there anything else I can try or look at? I would much prefer to use Engintron (or any kind of Nginx caching) if possible.

Thanks for your time.
